### ISSUE TITLE
CustomDataTypeTest fails in compiler

### ISSUE THREAD

**Issue by rmetzger** [stratosphere/stratosphere#255]
For example:
https://s3.amazonaws.com/archive.travis-ci.org/jobs/13633118/log.txt

```
Running eu.stratosphere.pact.test.contracts.io.CustomDataTypeTest
14:34:28,651 WARN  eu.stratosphere.pact.compiler.PactCompiler                    - Could not instantiate input format to obtain statistics. Limited statistics will be available.
java.lang.IllegalStateException: java.lang.ClassNotFoundException: eu.stratosphere.pact.test.external.TestClass
    at eu.stratosphere.nephele.configuration.Configuration.getClass(Configuration.java:115)
    at eu.stratosphere.pact.test.contracts.io.CustomDataTypeTest$EmptyInputFormat.configure(CustomDataTypeTest.java:95)
    at eu.stratosphere.pact.compiler.plan.DataSourceNode.computeOutputEstimates(DataSourceNode.java:123)
    at eu.stratosphere.pact.compiler.PactCompiler$GraphCreatingVisitor.postVisit(PactCompiler.java:975)
    at eu.stratosphere.pact.compiler.PactCompiler$GraphCreatingVisitor.postVisit(PactCompiler.java:771)
    at eu.stratosphere.pact.common.contract.GenericDataSource.accept(GenericDataSource.java:165)
    at eu.stratosphere.pact.common.contract.GenericDataSink.accept(GenericDataSink.java:376)
    at eu.stratosphere.pact.common.plan.Plan.accept(Plan.java:203)
    at eu.stratosphere.pact.compiler.PactCompiler.compile(PactCompiler.java:665)
    at eu.stratosphere.pact.compiler.PactCompiler.compile(PactCompiler.java:547)
    at eu.stratosphere.pact.test.contracts.io.CustomDataTypeTest.getJobGraph(CustomDataTypeTest.java:75)
    at eu.stratosphere.pact.test.util.TestBase.testJob(TestBase.java:105)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:606)
    at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
    at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
    at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
    at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
    at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:28)
    at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:31)
    at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:76)
    at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
    at org.junit.runners.ParentRunner$3.run(ParentRunner.java:193)
    at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:52)
    at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:191)
    at org.junit.runners.ParentRunner.access$000(ParentRunner.java:42)
    at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:184)
    at org.junit.runners.ParentRunner.run(ParentRunner.java:236)
    at org.junit.runners.Suite.runChild(Suite.java:128)
    at org.junit.runners.Suite.runChild(Suite.java:24)
    at org.junit.runners.ParentRunner$3.run(ParentRunner.java:193)
    at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:52)
    at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:191)
    at org.junit.runners.ParentRunner.access$000(ParentRunner.java:42)
    at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:184)
    at org.junit.runners.ParentRunner.run(ParentRunner.java:236)
    at org.apache.maven.surefire.junit4.JUnit4TestSet.execute(JUnit4TestSet.java:104)
    at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:147)
    at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:98)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:606)
    at org.apache.maven.surefire.booter.ProviderFactory$ClassLoaderProxy.invoke(ProviderFactory.java:101)
    at com.sun.proxy.$Proxy0.invoke(Unknown Source)
    at org.apache.maven.surefire.booter.SurefireStarter.invokeProvider(SurefireStarter.java:139)
    at org.apache.maven.surefire.booter.SurefireStarter.runSuitesInProcess(SurefireStarter.java:82)
    at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:81)
Caused by: java.lang.ClassNotFoundException: eu.stratosphere.pact.test.external.TestClass
    at java.net.URLClassLoader$1.run(URLClassLoader.java:366)
    at java.net.URLClassLoader$1.run(URLClassLoader.java:355)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.net.URLClassLoader.findClass(URLClassLoader.java:354)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
    at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
    at java.lang.Class.forName0(Native Method)
    at java.lang.Class.forName(Class.java:270)
    at eu.stratosphere.nephele.configuration.Configuration.getClass(Configuration.java:113)
    ... 49 more
14:34:28,875 WARN  eu.stratosphere.nephele.jobmanager.splitassigner.InputSplitManager  - Unable to find specific input split provider for type eu.stratosphere.nephele.template.GenericInputSplit, using default assigner
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 5.793 sec
```

I will look into the maven configuration to properly trigger a failed test.

**Comment by rmetzger** [comment]
The exception is being catched in the `DataSourceNode` class (Line 125)

**Comment by StephanEwen** [comment]
Okay, the problem is that the compiler has currently no user code class loader. I'll take a quick stab at that...

**Comment by StephanEwen** [comment]
First part is fixed in 80660c6e9dbc65802c6c53ba2f6275d20d98dbe4.

Next step is to provide the compiler with a PlanWithJars instead of only a Plan.

**Comment by rmetzger** [comment]
Test has been removed.
